packages/react-instantsearch-hooks-server/src/getServerState.tsx

```diff
@@ -8,19 +8,21 @@ import {
 import type { InitialResults, InstantSearch, UiState } from 'instantsearch.js';
 import type { IndexWidget } from 'instantsearch.js/es/widgets/index/index';
 import type { ReactNode } from 'react';
-import type { renderToString as reactRenderToString } from 'react-dom/server';
 import type {
   InstantSearchServerContextApi,
   InstantSearchServerState,
 } from 'react-instantsearch-hooks';
 
 type SearchRef = { current: InstantSearch | undefined };
 
+export type RenderToString = (element: JSX.Element) => unknown;
```

Comment:
```suggestion
export type RenderToString = (element: JSX.Element) => string;
```

Reply:
I left unknown as you should be able to use this argument with renderToStream or something as well, because we don't use the string

Reply:
`renderToReadableStream()` will probably create issues for example. But as long as its typed as a synchronous method it should be fine indeed.

Reply:
I was more thinking if there's in theory another method that traverses the tree, but doesn't render, it is also sufficient for this method